src/TableStorage.Abstractions/Store/TableStore.cs

```diff
@@ -69,9 +69,15 @@ public TableStore(string tableName, string storageConnectionString, TableStorage
             if (options.EnsureTableExists)
             {
 #if NETCOREAPP2_0 || NETCOREAPP2_1
-                CreateTableAsync().Wait();
+                if(!TableExistsAsync().Result)
+                {
+                    CreateTableAsync().Wait();
```

Comment:
CreateTableAsync and CreateTable call CreateIfNotExists which makes this change unnecessary.

Reply:
In appearance, yes, but in the background, we get 409 HTTP error. And for example in Azure Application Insights, I see it in the logs. Please see the code below. 
Logs can be filtered. But we can just check is table exist.

` public virtual bool CreateIfNotExists(TableRequestOptions requestOptions = null, OperationContext operationContext = null)
    {
      try
      {
        this.Create(requestOptions, operationContext);
        return true;
      }
      catch (StorageException ex)
      {
        if (ex.RequestInformation.HttpStatusCode == 409 && (ex.RequestInformation.ExtendedErrorInformation == null || ex.RequestInformation.ExtendedErrorInformation.ErrorCode == TableErrorCodeStrings.TableAlreadyExists))
          return false;
        throw;
      }
    }`

Reply:
Thanks for clarifying the issue